x/evm/keeper/state_transition.go

```diff
@@ -310,15 +312,17 @@ func (k *Keeper) ApplyMessageWithConfig(
 		stateDB.PrepareAccessList(msg.From(), msg.To(), evm.ActivePrecompiles(rules), msg.AccessList())
 	}
 
+	scaledMsgValue := utils.ConvertTo6Decimals(*msg.Value())
```

Comment:
shouldn't we scale the call `value` for internal EVM calls too? For example, if within a contract we send some funds to another address  

```
(bool sent, ) = another_address.call{value: 1000000000000000000}(""); // send 1 eth/evmos
```

We can achieve this by adding the corresponding conversion on the stateDB implementation (e.g `SetBalance`, `GetBalance` funcs)

Reply:
Let me test this out today and will add it

Reply:
@GAtom22 while this makes sense to me won't the values being passed into these internal calls be scaled down to 6 decimals by default already? Since the balances will be in uatom sending in 18 decimals will just lead to internal errors of insufficient balance ?

Reply:
Haven't tested yet, but for example, in this SetBalance function, if the `amount` comes with 18 decimals, the delta will be bigger than the actual

```
// SetBalance update account's balance, compare with current balance first, then decide to mint or burn.
func (k *Keeper) SetBalance(ctx sdk.Context, addr common.Address, amount *big.Int) error {
	cosmosAddr := sdk.AccAddress(addr.Bytes())

	params := k.GetParams(ctx)
	coin := k.bankKeeper.GetBalance(ctx, cosmosAddr, params.EvmDenom)  // This will be with 6 decimals
	balance := coin.Amount.BigInt()
	delta := new(big.Int).Sub(amount, balance)
```

Reply:
@GAtom22 I scaled up the `SetBalance` function that was indeed minting excessively. Nice catch 🙌🏻 . For the other concern of internal calls, it still fails the balance check if the contract or EOA doesn't have enough to process a send and it fails gracefully.